backend/server/server.go

```diff
@@ -22,35 +28,52 @@ var (
 type UIServer struct {
 	ui.UnimplementedUIServer
 
-	hubbleAddr   string
-	hubbleClient observer.ObserverClient
+	relayAddr       string
+	relayConnParams *grpc.ConnectParams
+	hubbleClient    observer.ObserverClient
+	grpcConnection  *grpc.ClientConn
+	retryPolicy     retries.BackOff
 
 	k8s       kubernetes.Interface
 	dataCache *dataCache
 }
 
-func New(hubbleAddr string) *UIServer {
+func New(relayAddr string) *UIServer {
+
 	return &UIServer{
-		hubbleAddr: hubbleAddr,
-		dataCache:  newDataCache(),
+		relayAddr: relayAddr,
+		relayConnParams: &grpc.ConnectParams{
+			Backoff: backoff.Config{
+				BaseDelay:  1.0 * time.Second,
```

Comment:
these need to be configurable. we can do something similar to relay: https://github.com/cilium/cilium/blob/master/pkg/hubble/relay/pool/option.go